sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetReadState.java

```diff
@@ -131,9 +157,10 @@ long currentRangeEnd() {
    * Advance the current offset and rowId to the new values.
    */
   void advanceOffsetAndRowId(int newOffset, long newRowId) {
-    valuesToReadInBatch -= (newOffset - offset);
+    rowsToReadInBatch -= (newOffset - levelOffset);
```

Comment:
Hmm, for repeated values, is offset difference the same as row difference? I think one row could contain many values (offsets), no?

Reply:
Yes, again this is not called when reading repeated values. I probably should remove this method and directly operate on these fields in `readBatchInternal`.